### PR TITLE
(SERVER-3157) Add Snyk GH Action

### PR DESCRIPTION
**Pull request by Magisus**

### CODE REVIEW

**Comment by Magisus** [comment]
This doesn't seem to be working right yet.

**Comment by Magisus** [comment]
For reasons that are entirely unclear to me, using the `pull_request_target` trigger does not appear to work. `pull_request` does.

**Comment by Magisus** [comment]
Update: apparently you have to merge actions using the `pull_request_target` trigger before they will run. So I'm going to put this to `pull_request` for now so we can see it go, then flip to `pull_request_target` right before we merge.

**Review by rileynewton — CHANGES_REQUESTED** [review]
I found out from Jeremy that there's a specific puppet-foss token (there's a token for every org now I realize 🤦 ), you'll likely want to switch to that.

**Comment by Magisus** [comment]
I asked him about that several times and he told me there was only one for anything that goes into PE...

File: .travis.yml [hunk 1]
```diff
@@ -21,9 +21,6 @@ jobs:
     - jdk: openjdk11
       env: MULTITHREADED=true
       name: "Java 11 w/ multithreaded"
-    - stage: puppetserver vulnerability scan
-      script:
-        - lein nvd check
```
Comment (mwaggett) [inline .travis.yml:26]: Should we also remove the plugin from project.clj and remove the suppression file?

**Review by mwaggett — APPROVED** [review]

**Comment by Magisus** [comment]
Flipping this to `pull_request_target` now that Snyk has passed

**Review by mwaggett — APPROVED** [review]

**Pull request merged**